### ISSUE TITLE
Remove run directories that are no longer needed by other tests during the regression tests

### ISSUE THREAD

**Issue by binli2337** [ufs-community/ufs-weather-model#1003]
## Description

During regression tests, many run directories are created and those run directories are removed only after all the tests are completed. To reduce disk space used during the tests, the "run_test.sh" script needs to be updated to remove run directories that are no longer needed by other tests during the regression tests.

## Solution
Revise run_test.sh.
